# **2 – Visual Study**

## Objectives

* Explore the cherry leaf image dataset visually
* Differentiate between healthy and powdery mildew-infected leaves
* Support Business Requirement #1 through image analysis and insights

## Inputs

* Images located in -  inputs/dataset/raw/cherry-leaves/
* Subfolders - healthy/, powdery_mildew/

## Outputs

* Image count per class
* Sample images from each class
* Average and variability images
* Images displayed to illustrate visual patterns




--- 

# Change working directory

Change the working directory from its current folder to its parent folder

In [1]:
import os
current_dir = os.getcwd()
current_dir

'c:\\Users\\amyno\\OneDrive\\Documents\\CherryLeafProject\\milestone-project-mildew-detection-in-cherry-leaves\\jupyter_notebooks'

Make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chir() defines the new current directory

In [2]:
os.chdir(os.path.dirname(current_dir))
print("You set a new current directory")

You set a new current directory


Confirm the new current directory

In [3]:
current_dir = os.getcwd()
current_dir

'c:\\Users\\amyno\\OneDrive\\Documents\\CherryLeafProject\\milestone-project-mildew-detection-in-cherry-leaves'

---

# Count of images per class


In this section, I count how many images exist for each class (healthy vs powdery mildew).
This helps confirm the dataset is balanced and gives context for later modeling.

Set path to image folders

In [4]:
image_path = "inputs/dataset/raw/cherry-leaves"
class_names = os.listdir(image_path)

Determine no. of images per class

In [5]:
for class_name in class_names:
    class_dir = os.path.join(image_path, class_name)
    image_count = len(os.listdir(class_dir))
    print(f"{class_name}: {image_count} images")

healthy: 2104 images
powdery_mildew: 2104 images


This shows that there is the same amount of healthy and diseased images to analyse. This is important as it shows that the data sets are even and there is no skew in favour of either class. 

It also shows that there is a healthy amount of data to work with. 2104 images per class allows more than enough data to train, test and validate the model.